src/coreclr/jit/importercalls.cpp

```diff
@@ -2589,6 +2589,9 @@ GenTree* Compiler::impIntrinsic(GenTree*                newobjThis,
             case NI_System_Threading_Interlocked_MemoryBarrier:
             case NI_System_Threading_Interlocked_ReadMemoryBarrier:
 
+            case NI_System_Threading_Volatile_Read:
+            case NI_System_Threading_Volatile_Write:
+
```

Comment:
should it be "always expand" instead of "betterToExpand" ?

Reply:
There's a managed fallback in the runtime (which is always used for long on 32bit). Not sure whether it makes sense to change this to never use that.